app/api/services/core/patch_params.py

```diff
@@ -0,0 +1,12 @@
+import dataclasses
+from typing import Generic, TypeVar
+
+ResourceType = TypeVar("ResourceType")
+
+
+@dataclasses.dataclass
+class PatchParams(Generic[ResourceType]):
+    """Parameters for patching a resource."""
+
+    resource: ResourceType
+    fields_to_patch: list[str]
```

Comment:
I wonder if a utility generator method would be convenient here, something like:
```py
def get_fields_to_patch(self) -> Iterable[Tuple[str, Any]]:
     for field in self.fields_to_patch:
          value = getattr(self.resource, field)
          yield field, value
```
Might make the typing annoying, but I'm not sure how the existing approach doesn't hit typing issues.

Reply:
i like that idea! could i create a ticket to do that in a future PR?

Reply:
Created this ticket https://github.com/navapbc/template-application-flask/issues/61

Reply:
As part of the last commit I went ahead and did this in a function called `fields_to_patch`, it's not an instance method though, i ended up trying one of @zelgadis 's earlier ideas of using a "unset"/"missing" sentinal value. (looks like under the hood Marshmallow does something similar, and I wouldn't be surprised if Pydantic also has something similar). thoughts on this approach? i like how explicit it is because then it makes it possible and clear to do things like

```
class FooParams:
    non_null: str | Missing
    can_be_null: str | None | Missing
```